engine/cache/mountsync.go

```diff
@@ -91,11 +91,23 @@ func (m *manager) StartCacheMountSynchronization(ctx context.Context) error {
 
 	m.stopCacheMountSync = func(ctx context.Context) error {
 		var eg errgroup.Group
+
+		allCacheMounts := map[string]struct{}{}
+		core.SeenCacheKeys.Range(func(k any, v any) bool {
+			allCacheMounts[k.(string)] = struct{}{}
+			return true
+		})
+
 		for _, syncedCacheMount := range syncedCacheMounts {
```

Comment:
do we still want to support manually synced mounts? I guess that's for backward compat / avoiding regressions. Should probably put a fat warning that we should remove this?

Reply:
actually now that you mention about it I think we can remove this altogether without any regressions. IIUC the current code will always  call the `stopCacheMountSync` regardless if the volume if used or not. Now that we can detect which volumes the engine ever used during its lifetime, seems more accurate to only sync those. WDYT?